src/driveshaft-config.cpp

```diff
@@ -0,0 +1,269 @@
+    for (auto i = servers_list.begin(); i != servers_list.end(); ++i) {
+        if (!i->isString()) {
+            LOG4CXX_ERROR(MainLogger, cfgkeys::GEARMAN_SERVERS_LIST << " does not contain strings");
+            throw std::runtime_error("config server list type failure");
+        }
+
+        const auto &name = i->asString();
+        this->m_server_list.insert(name);
+        LOG4CXX_DEBUG(MainLogger, "Read server: " << name);
+    }
+}
+
+void DriveshaftConfig::parsePoolList(const Json::Value &node) {
+    const auto &pools_list = node[cfgkeys::POOLS_LIST];
+    for (auto i = pools_list.begin(); i != pools_list.end(); ++i) {
+        const auto &pool_name = i.name();
+        const auto &pool_node = *i;
+
+        if (!pool_node.isMember(cfgkeys::POOL_WORKER_COUNT) ||
+            !pool_node.isMember(cfgkeys::POOL_JOB_LIST) ||
+            !pool_node.isMember(cfgkeys::POOL_JOB_PROCESSING_URI) ||
+            !pool_node[cfgkeys::POOL_WORKER_COUNT].isUInt() ||
+            !pool_node[cfgkeys::POOL_JOB_LIST].isArray() ||
+            !pool_node[cfgkeys::POOL_JOB_PROCESSING_URI].isString()) {
+            LOG4CXX_ERROR(
+                MainLogger,
+                "Config (" << m_config_filename << ") has invalid " <<
+                cfgkeys::POOLS_LIST << " elements: (" <<
+                cfgkeys::POOL_WORKER_COUNT << ", " <<
+                cfgkeys::POOL_JOB_PROCESSING_URI << ", " <<
+                cfgkeys::POOL_JOB_LIST << ")"
+            );
+
+            throw std::runtime_error("config jobs list parse failure");
+        }
+
+        auto &pool_data = this->m_pool_map[pool_name];
+        pool_data.worker_count = pool_node[cfgkeys::POOL_WORKER_COUNT].asUInt();
+        pool_data.job_processing_uri = pool_node[cfgkeys::POOL_JOB_PROCESSING_URI].asString();
+        LOG4CXX_DEBUG(
+            MainLogger,
+            "Read pool: " << pool_name << " with count " <<
+            pool_data.worker_count << " and URI " << pool_data.job_processing_uri
+        );
+
+        const auto &job_list = pool_node[cfgkeys::POOL_JOB_LIST];
+        for (auto j = job_list.begin(); j != job_list.end(); ++j) {
+            if (!j->isString()) {
+                LOG4CXX_ERROR(MainLogger, cfgkeys::POOL_JOB_LIST << " does not contain strings");
+            }
+
+            const auto &name = j->asString();
+            pool_data.job_list.insert(name);
+            LOG4CXX_DEBUG(MainLogger, "Pool " << pool_name << " adding job " << name);
+        }
+    }
+}
+
+bool DriveshaftConfig::needsConfigUpdate(const std::string &new_config_filename) const {
+    boost::filesystem::path config_path(new_config_filename);
+    std::time_t modified_time = boost::filesystem::last_write_time(config_path);
+    if (modified_time == (std::time_t)-1) {
+        LOG4CXX_ERROR(
+            MainLogger,
+            "Unable to stat file " << new_config_filename <<
+            ". Error: " << std::to_string(errno)
```

Comment:
I thought I read somewhere that (std::time_t - 1) was the value returned from a bad stat() in [last_write_time](http://www.boost.org/doc/libs/1_46_0/libs/filesystem/v3/doc/reference.html#last_write_time). After re-reading the documentation, I now no longer remember where I saw that.